app/Mail/VoucherPaymentRequestEmail.php

```diff
@@ -28,7 +28,7 @@ public function __construct($user, $trader, $vouchers, $file)
         // Sending vouchers collection in case we want more than just count in email for copy.
         $this->vouchers = $vouchers;
         $this->file = $file;
-        $this->market = $trader->market;
+        $this->market = $trader->market->name;
```

Comment:
We could probably ditch the market from here altogether. And get from Trader when it is used. Also we have not yet forced the requirement - so might be better as ternary with a default?